src/containers/PageContainer/elements/Header/elements/ConnectWalletButton/hooks/useConnectWalletButtonData.ts

```diff
@@ -0,0 +1,49 @@
+import { useMemo } from 'react';
+import { AbstractConnector } from '@web3-react/abstract-connector';
+import { LOCAL_STORAGE_KEYS } from 'constants/localStorage';
+
+type UseConnectWalletButtonDataProps = {
+	props: {
+		isDesktop: boolean;
+		account: string | null | undefined;
+		active: boolean;
+		connector: AbstractConnector | undefined;
+	};
+};
+
+type UseConnectWalletButtonDataReturn = {
+	formattedData: {
+		wallet: string | null;
+		isConnected: boolean;
+		connectTitle: string;
+		disconnectTitle: string;
+		isConnecting: boolean;
+	};
+};
+
+export const useConnectWalletButtonData = ({
+	props,
+}: UseConnectWalletButtonDataProps): UseConnectWalletButtonDataReturn => {
+	const formattedData = useMemo(() => {
+		const wallet = localStorage.getItem(LOCAL_STORAGE_KEYS.CHOOSEN_WALLET);
+		const isConnected =
+			props.active && Boolean(props.account) && Boolean(props.connector);
+		const isConnecting = !isConnected && Boolean(wallet);
+		const connectTitle = isConnecting
+			? `Trying to connect ${wallet}`
+			: `Connect ${props.isDesktop && 'Wallet'}`;
```

Comment:
Did you want to remove all hardcoded values and add as constants?

Reply:
Hmm, this title was defined in `JSX` and I moved it into custom hooks. Not sure it is constant, it is detecting desktop or mobile, and set the title.